xla/service/gpu/fusions/triton/triton_fusion_emitter_legacy_matmul.cc

```diff
@@ -1206,8 +1229,18 @@ class MatMulEmitterHelper {
     }
 
     auto add_dim = [&](const DimProperties& properties) -> absl::Status {
-      if (analysis_.IterSpec(side.scope, hlo, properties.index) == nullptr) {
-        return absl::OkStatus();
+      {
+        const TensorIterationSpec::DimIterationSpec* spec =
+            analysis_.IterSpec(side.scope, hlo, properties.index);
+        if (spec == nullptr ||
+            (IsNonTrivialTiledDimension(side.scope, properties.index) &&
+             spec->size() == 1 && spec->at(0).count == 1)) {
+          // If a non-trivial tiled dimension has a trivial iteration spec at
```

Comment:
Can you define formally what a trivial iteration spec is? Is it a spec with size 1 and count 1?

Maybe make a helper `IsTrivialIterationSpec` that documents the expectations & meaning?

Reply:
I'll reformulate to say that the dimension has only one element to avoid new terms then.